### PR TITLE
Feature/594 runtime env configs

### PR DESCRIPTION
**Pull request by diablouma**
* Using multi stage docker build for the dashboard
* Using runtime environment variables to configure if the enable/disable the OAUTH for the dashboard
* Use LHD_OAUTH_ENABLED as the config name instead of AUTHENTICATION_ENABLED

### CODE REVIEW

File: dashboard/Dockerfile
Comment (coltmcnealy-lh) [inline dashboard/Dockerfile:1]: Could we move this to the `/build` directory in the server? Just trying to standardize everything.

File: docker/dashboard/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,25 @@
-FROM node:20
+FROM node:20 AS base
```
Comment (coltmcnealy-lh) [inline docker/dashboard/Dockerfile:1]: Ah, I see this is here. Could we delete the other `Dockerfile` then and just use this one? The rest of the repo only has dockerfiles here.

**Review by coltmcnealy-lh — APPROVED** [review]
Approved but please address the Dockerfile comments before merging.

File: docker/dashboard/Dockerfile (hunk 1 above)
Comment (diablouma) [reply docker/dashboard/Dockerfile:1]: you are right @coltmcnealy-lh I forgot to delete the other file.

File: dashboard/Dockerfile
Comment (diablouma) [reply dashboard/Dockerfile:1]: just deleted it, as it should be under the docker folder

**Pull request merged**